### PR TITLE
Refactor all Kafka producers/consumers to use a standardized set of Clap-parsed env vars

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
None

### What changes does this PR make to Grapl? Why?
Refactor all Kafka producers/consumers to use a standardized set of Clap-parsed env vars.
They're encapsulated in KafkaProducerConfig and KafkaConsumerConfig.
Call ::parse on these to retrieve from env.
This was originally introduced with Graph Dispatcher; I've pulled it out to the Kafka library. 

### How were these changes tested?
make test-integration-new

### CODE REVIEW

File: src/rust/kafka/src/lib.rs [hunk 1]
```diff
@@ -290,6 +324,18 @@ where
         })
     }
 
+    pub fn new_from_config(
```
Comment (jgrillo-grapl) [inline src/rust/kafka/src/lib.rs:324]: same

**Review by jgrillo-grapl — APPROVED** [review]
Looks good, I think we should break the existing constructor and replace it with this new one. Not going to block on that, though.

File: src/rust/generators/sysmon-generator/src/main.rs [hunk 2]
```diff
@@ -79,7 +79,7 @@ async fn handler() -> Result<(), SysmonGeneratorError> {
 
     // TODO: also construct a stream processor for retries
 
-    let stream_processor = StreamProcessor::new_from_config(
+    let stream_processor = StreamProcessor::new(
         consumer_config,
         consumer_topic,
```
Comment (jgrillo-grapl) [inline src/rust/generators/sysmon-generator/src/main.rs:84]: The topics should probably go in the config object also, sorry I missed this before.

Comment (colin-grapl) [reply src/rust/generators/sysmon-generator/src/main.rs:84]: Agreed. NBD / can be done later, but if we place the topic itno the config we could also change this `new_from_config` into a TryFrom<Config>

**Review by christophermaier — APPROVED** [review]
Nice overall cleanup 👍 

I have some minor quibbles about naming, etc., but no blockers.

File: src/rust/generators/sysmon-generator/tests/integration_test.rs [hunk 3]
```diff
@@ -3,8 +3,10 @@
 use std::time::Duration;
 
 use bytes::Bytes;
+use clap::Parser;
 use futures::StreamExt;
 use kafka::{
+    config::KafkaConsumerConfig,
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply src/rust/generators/sysmon-generator/tests/integration_test.rs:9]: yeah, this is mostly a legacy from when it lived in a different crate

**Review by jgrillo-grapl — APPROVED** [review]
LGTM

**Pull request merged**